packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -175,6 +166,51 @@ export function canonicalizeEntryConfig(
   };
 }
 
+export function canonicalizeBoundsCheckConfig(
+  boundsCheckConfig: GPCProofEntryBoundsCheckConfig
+): GPCProofEntryBoundsCheckConfig {
+  const { inRange, notInRange } = boundsCheckConfig;
+
+  // Throw if an invalid interval is specified to avoid invalid
+  // canonicalisations.
+  for (const interval of [inRange, notInRange]) {
+    if (interval && interval.min > interval.max) {
+      throw new Error(
+        `Invalid bounds check interval ${interval} in config cannot be canonicalized. The minimum value must be less than or equal to the maximum value.`
+      );
+    }
+  }
+
+  return {
+    ...(!inRange && !notInRange
+      ? {}
+      : !inRange
+      ? { notInRange: notInRange }
+      : !notInRange
+      ? { inRange: inRange }
+      : // inRange\notInRange = [inRange.min, ⍵] for some ⍵.
```

Comment:
Does "\" mean intersection in this case?  Or set difference?  I'm not familiar with the usage.